### ISSUE TITLE
Requests Should Show Actual Items

### ISSUE THREAD

**Issue by seanmarcia** [rubyforgood/partner#100]
Currently when a partner makes a request is only shows the categories that the diaper bank carries not the specific items.
Partners should be able to request specific items not just categories.

* This will require work on both apps

**Comment by willmurphyscode** [comment]
I wanted to fill in a couple of the details on 

> This will require work on both apps

We will need to update https://github.com/rubyforgood/partner/blob/099fa5f4bc148d18b65df0ce1ea4beae497dd38f/app/models/partner_request.rb#L9 to make it send the additional information, and simultaneously update https://github.com/rubyforgood/diaper/blob/5fbae3ffbd0eb76e49f2193d9ced1f1f295c8225/app/controllers/api/v1/partner_requests_controller.rb#L10 to parse the changed request correctly.

Should we make separate issues for those two changes?

**Comment by seanmarcia** [comment]
Hey @adam-barton and @willmurphyscode is the associated PR ready to be reviewed? Sorry I've been out of touch and wanted to check the status on this.

**Comment by seanmarcia** [comment]
I've started looking into this and will work on it this weekend. I think, based on my initial examinations of how stuff works, that I'm going to change and simplify the `formatted_item_requests_hash` method in the `partner_requests.rb` model to send an array of hashes. Probably something like:
```
   def formatted_item_requests_hash(item_requests)
     item_requests.map { |item| { "item_id" => item.item_id, "quantity" => item.quantity } }
   end
```

I'll obviously need to update things on the diaper app side but since the diaper app stores the `request_items` as jsonb it shouldn't be a problem to ship it over like this.

It would probably simplified the controller on the diaper size a bit too. We could change ` request_items: BaseItem.pluck(:partner_key).map(&:to_sym)` to `request_items: [:item_id, :quantity]`. This will obviously require refactoring to switch from a hash to array of hashes.

Is this a path you went down at RbtB and it didn't work or am I missing something else not obvious before I work on this? I was a little concerned it might affect the family details but it looks like kevin has it fully separated out (endpoints, controllers, etc.) I know on the diaper side we'll need to figure out how to not break existing requests which may be why this isn't how things were implemented.

/cc @willmurphyscode @adam-barton @joshleichtung @armahillo
